paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java

```diff
@@ -210,11 +212,31 @@ public void createTable(ObjectPath tablePath, CatalogBaseTable table, boolean ig
         }
 
         // remove table path
-        String specific = options.remove(PATH.key());
-        if (specific != null) {
-            catalogTable = catalogTable.copy(options);
+        options.remove(PATH.key());
+
+        // Put Paimon catalog global options into table options
+        // ref https://github.com/apache/incubator-paimon/issues/754
+        if (catalog instanceof org.apache.paimon.catalog.AbstractCatalog) {
```

Comment:
We should put this into `AbstractCatalog`. Other engines should have this capability too.
In `AbstractCatalog`:
```
protected AbstractCatalog(FileIO fileIO, Map<String, String> catalogOptions) {
        this.fileIO = fileIO;
        this.tableDefaultOptions = ...;
}
```

And copy Schema in `createTable` with table default options.